app/models/ems_refresh/save_inventory_infra.rb

```diff
@@ -159,6 +159,10 @@ def save_hosts_inventory(ems, hashes, target = nil)
           #   and the new hostname is not an ip address
           h[:name] = h[:hostname] if h[:name] =~ ip_part && !(h[:hostname] =~ ip_whole)
 
+          # Update the name to be found.name if it has the same ems_ref and the name
+          # already has a '- int' suffix to work around duplicate hostnames
+          h[:name] = found.name if found.name =~ /#{h[:name]} - \d+$/ && h[:ems_ref] == found.ems_ref
```

Comment:
Yes that is the case `h[:ems_ref] == found.ems_ref` is handling.  I was thinking if we were re-using an old host record we would want to re-evaluate what the name should be, though I don't think there'd be any downside other than possibly using a larger suffix.